internal/dataplane/kong_client.go

```diff
@@ -423,21 +424,29 @@ func (c *KongClient) Update(ctx context.Context) error {
 // enables filtering for which objects are currently applied to the data-plane,
 // as well as updating the c.kubernetesObjectStatusQueue to queue those objects
 // for reconciliation so their statuses can be properly updated.
-func (c *KongClient) triggerKubernetesObjectReport(objs ...client.Object) {
+func (c *KongClient) triggerKubernetesObjectReport(succeededObjs []client.Object, parsingErrors []parser.ParsingError) {
 	// first a new set of the included objects for the most recent configuration
 	// needs to be generated.
 	set := k8sobj.Set{}
-	for _, obj := range objs {
-		set.Insert(obj)
+	for _, obj := range succeededObjs {
+		set.InsertSucceeded(obj)
 	}
 
 	c.updateKubernetesObjectReportFilter(set)
 
+	failedObjs := make([]client.Object, 0, len(parsingErrors))
+	for _, err := range parsingErrors {
+		failedObjs = append(failedObjs, err.RelatedObjects()...)
+		set.InsertFailed(err.RelatedObjects()...)
+	}
```

Comment:
The filter gets propagated both for succeeded and failed objects. It's probably worth changing its name, I left it as it was though for now.